src/foolscap/test/test_appserver.py

```diff
@@ -110,17 +110,17 @@ def test_bad_version(self):
         f = open(os.path.join(basedir, "services.json"), "wb")
         json.dump(orig, f)
         f.close()
-        e = self.failUnlessRaises(server.UnknownVersion,
-                                  server.load_service_data, basedir)
-        self.failUnlessIn("unable to handle version 99", str(e))
+        e = self.assertRaises(server.UnknownVersion,
```

Comment:
Note to self: See which `assertRaises` this is calling (stdlib or trial or something else)

Reply:
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises

Reply:
I've almost always used `assertRaises` as a context-manager (i.e. using `with`) but i'm not clear on the difference between that and not ..

Reply:
It can be used without the context-manager also.

Example.

```
import unittest
import operator

class MyTest(unittest.TestCase):

    def testAssertRaises(self):
        self.assertRaises(ZeroDivisionError, operator.floordiv, 1, 0)
```

Reply:
Resolving.